include/tsl/sparse_hash.h

```diff
@@ -1833,13 +1833,9 @@ class sparse_hash : private Allocator,
   template <class K, class... Args>
   std::pair<iterator, bool> insert_impl(const K &key,
                                         Args &&...value_type_args) {
-    if (size() >= m_load_threshold_rehash) {
+    if (m_sparse_buckets_data.empty()) {
```

Comment:
These lines may not be needed. An empty `m_sparse_buckets_data` means that `m_sparse_buckets` points to `static_empty_sparse_bucket_ptr` and we will thus reach the later rehash call (just a bit slower on the first insertion but worth it for the simplicity).